src/test/ui/issues/issue-13847.rs

```diff
@@ -1,3 +1,3 @@
 fn main() {
-    return.is_failure //~ ERROR no field `is_failure` on type `!`
+    return.is_failure //~ ERROR no field `is_failure` on type `()`
```

Comment:
Haven't looked much at the implementation here, but this change alone signals something is probably not correct with this PR; the type of `return` *should* be `!`, not `()`